packages/frontend/src/state/lp/hooks.ts

```diff
@@ -68,6 +69,62 @@ export const useOpenPositionDeposit = () => {
   return openPositionDeposit
 }
 
+// Close position with flashloan
+export const useClosePosition = () => {
+  const address = useAtomValue(addressAtom)
+  const controllerHelperContract = useAtomValue(controllerHelperHelperContractAtom)
+  const controllerContract = useAtomValue(controllerContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getDebtAmount = useGetDebtAmount()
+  const getVault = useGetVault()
+  const getPosition = useGetPosition()
+  const getQuote = useGetQuote()
+  const closePosition = useAppCallback(async (vaultId: number, onTxConfirmed?: () => void) => {
+    const vaultBefore = await getVault(vaultId)
+    const uniTokenId = vaultBefore?.NFTCollateralId 
+    const position = await getPosition(uniTokenId)
+
+    if (
+      !controllerContract ||
+      !controllerHelperContract ||
+      !address ||
+      !position ||
+      !vaultBefore ||
+      !vaultBefore.shortAmount
+    )
+      return
+
+      const shortAmount = fromTokenAmount(vaultBefore.shortAmount, OSQUEETH_DECIMALS)
+      const debtInEthPromise = getDebtAmount(shortAmount)
+      const limitEthPromise = getQuote(shortAmount, true)
+      const [debtInEth, limitEth] = await Promise.all([debtInEthPromise, limitEthPromise])
+      const collateralToFlashloan = debtInEth.multipliedBy(COLLAT_RATIO)
+
+    const flashloanCloseVaultLpNftParam = {
+      vaultId: vaultId,
+      tokenId: uniTokenId,
+      liquidity: position.liquidity,
+      liquidityPercentage: fromTokenAmount(1, 18).toFixed(0),
+      wPowerPerpAmountToBurn: shortAmount.toFixed(0),
+      collateralToFlashloan: collateralToFlashloan.toFixed(0),
+      collateralToWithdraw: 0,
+      limitPriceEthPerPowerPerp: limitEth,
```

Comment:
I think this param is incorrect and a couple of issues with how its being calculated.  

- This should be a limit price in units of ETHPerPowerPerp ie 0.1 ETH per 1 squeeth, but scaled by 1e18.
- The calculation is assuming that you are selling oSQTH, but this is not always the case, sometimes you have to buy, so you have to check if the debt being burned is > or < than the LP composition.
- Right now you are getting the amountOut from the swap (which is a reallllllly high limit price and will be sandwiched/always execute regardless of what market does)
- This also needs to use slippage somewhere a user should specify a slippage for this trade and that needs to be taken into account when the limit price is calculated.